app/repositories/ThrottlingRepository.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import config.MicroserviceAppConfig
+import config.featureswitch.FeatureSwitching
+import play.api.libs.json.{Format, JsObject, JsValue, Json}
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.commands.FindAndModifyCommand
+import reactivemongo.api.indexes.{Index, IndexType}
+import reactivemongo.bson.{BSONDocument, BSONObjectID}
+import reactivemongo.play.json.ImplicitBSONHandlers._
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+import java.time.Instant
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class InstantProvider @Inject()() {
+
+  def getInstantNowMilli: Long = Instant.now.toEpochMilli
+
+}
+
+@Singleton
+class ThrottlingRepository @Inject()(mongo: ReactiveMongoComponent,
+                                     appConfig: MicroserviceAppConfig,
+                                     instantProvider: InstantProvider)(implicit ec: ExecutionContext)
+
+  extends ReactiveRepository[JsObject, BSONObjectID](
+    "throttling",
+    mongo.mongoConnector.db,
+    implicitly[Format[JsObject]],
+    implicitly[Format[BSONObjectID]]
+  ) with FeatureSwitching {
+
+  val throttleIdKey = "throttleId"
+  val timecodeKey = "timecode"
+  val countKey = "count"
+  val lastUpdatedTimestampKey = "lastUpdatedTimestamp"
+
+  private def timecode(time: Long) = time / 60000
```

Comment:
I'm not a huge fan of "epoch-minutes" as a value because it isn't really human readable and thus is hard to debug.  On the other hand it's cheap as chips.

Reply:
I agree with it being cheap as chips :)
I have no intention of ever needing to read the value as an actual time, just a numeric value which goes up by 1 every minute.